yapapi/config.py

```diff
@@ -0,0 +1,45 @@
+from dataclasses import dataclass, field
+from functools import partial
+import os
+from typing import Optional
+
+
+class MissingConfiguration(Exception):
+    def __init__(self, key: str, description: str):
+        self._key = key
+        self._description = description
+
+    def __str__(self):
+        return f"Missing configuration for {self._description}. Please set env var {self._key}."
+
+
+@dataclass
+class ApiConfig:
+    """
+    Yagna low level API configuration
+    Attributes:
+        app_key: Yagna application key.
+            If not provided, the default is to get the value from `YAGNA_APPKEY` environment variable.
+            If no value will be found MissingConfiguration error will be thrown
+        api_url: base URL or all REST API URLs. Example value: http://127.0.10.10:7500 (no trailing slash).
+            Uses YAGNA_API_URL environment variable
+        market_url: If not provided `api_url` will be used to construct it.
+            Uses YAGNA_MARKET_URL environment variable
+        payment_url: If not provided `api_url` will be used to construct it.
+            Uses YAGNA_PAYMENT_URL environment variable
+        net_url: Uses If not provided `api_url` will be used to construct it.
+            YAGNA_NET_URL environment variable
+        activity_url: If not provided `api_url` will be used to construct it.
+            Uses YAGNA_ACTIVITY_URL environment variable
+    """
+
+    app_key: str = field(default_factory=partial(os.getenv, "YAGNA_APPKEY"))  # type: ignore
+    api_url: str = field(default_factory=partial(os.getenv, "YAGNA_API_URL", "http://127.0.0.1:7465"))  # type: ignore
```

Comment:
IDK why mypy thinks it is `Optional[str]` here

Reply:
well... because it _is_ optional ?

iow, os.getenv _can_ return `None` ...